### ISSUE TITLE
[Windows][macOS] AIR 33 doesn't play MP4 (H.264) video

### ISSUE THREAD

**Issue by itlancer** [airsdk/Adobe-Runtime-Support#80]
## Problem Description
Windows applications built with AIR 33.0.1.220 doesn't play MP4 (H.264) video.

Related issues with AIR 32:
https://forums.adobe.com/thread/2614962
https://tracker.adobe.com/#/view/AIR-4198861
https://tracker.adobe.com/#/view/AIR-4198832
https://tracker.adobe.com/#/view/AIR-4198831
https://tracker.adobe.com/#/view/AIR-4198829
https://tracker.adobe.com/#/view/AIR-4198826
https://tracker.adobe.com/#/view/AIR-4198824
https://tracker.adobe.com/#/view/AIR-4198823
https://tracker.adobe.com/#/view/AIR-4198822
https://tracker.adobe.com/#/view/AIR-4198819
https://tracker.adobe.com/#/view/AIR-4198798
https://tracker.adobe.com/#/view/AIR-4198834
https://tracker.adobe.com/#/view/AIR-4198835
https://tracker.adobe.com/#/view/AIR-4198863
https://tracker.adobe.com/#/view/AIR-4198864

Tested with multiple devices across many Windows 8.1 and Windows 10 OSes.
It has been tested with many AIR 32 beta versions and devices even with latests AIR 32.0.0.144 beta and AIR 33.0.1.220.
With AIR 33.0.1.220 MP4 (H.264) video doesn't work at all.
With AIR 32.0.0.144 beta video partially works but still have some problems: https://forums.adobe.com/message/11170701#11170701
It works fine with AIR 32.0.0.89.
`NetStream::useHardwareDecoder = true` with Windows doesn't help.
FLV (H.263) videos works fine in all these cases.

## Steps to Reproduce
Launch code below. Application example with sources and example of video attached.
[air33_video.zip](https://github.com/Gamua/Adobe-Runtime-Support/files/3520514/air33_video.zip)

```
package  {
	import flash.display.Sprite;
	import flash.events.Event;
	import flash.net.NetConnection;
	import flash.net.NetStream;
	import flash.media.Video;
	import flash.events.NetStatusEvent;
	
	public class AIR33Video extends Sprite {
		private var video:Video = new Video(640, 480);
		private var nc:NetConnection;
		
		public function AIR33Video() {
			addEventListener(Event.ADDED_TO_STAGE, init);
		}
		
		private function init(e:Event):void {
			removeEventListener(Event.ADDED_TO_STAGE, init);
			
			addChild(video);
			
			nc = new NetConnection();
			nc.addEventListener(NetStatusEvent.NET_STATUS, ncHandler);
			nc.connect(null);
		}
		
		private function ncHandler(e:NetStatusEvent):void {
			if (e.info.code == "NetConnection.Connect.Success"){
				var ns:NetStream = new NetStream(nc);
				ns.client = {onMetaData:getMeta};
				ns.addEventListener(NetStatusEvent.NET_STATUS, nsHandler);
				video.attachNetStream(ns);
				ns.play("video.mp4");
			}
		}
		
		private function nsHandler(e:NetStatusEvent):void {
			trace(e.info.code);
		}
		
		private function getMeta(mdata:Object):void { }
		
	}
}
```

**Actual Result:**
Video doesn't play. NetStream fires `NetStream.Play.Failed` event.

**Expected Result:**
Video playback.

## Known Workarounds
none (use AIR 32.0.0.89)

**Comment by itlancer** [comment]
With AIR 33.0.2.338 some videos playback via `Video`, `StageVideo` and `VideoTexture`, but some videos doesn't work at all (earlier it works fine with AIR 32.0.0.89).
Such video example: 
[video.zip](https://github.com/Gamua/Adobe-Runtime-Support/files/3963954/video.zip)
Seems that video with audio doesn't work.
Also video playback with `NetStream::useHardwareDecoder=false` still doesn't work: https://github.com/Gamua/Adobe-Runtime-Support/issues/155

**Comment by itlancer** [comment]
The same problem with AIR 33.0.2.338 and AIR 33.1.0.16 beta for macOS.

**Comment by itlancer** [comment]
Issue still exists with latest AIR 33.1.1.50.

**Comment by bobsmo** [comment]
issue still exists with latest AIR 33.1.1.86

**Comment by jasonJamEther** [comment]
This is a really crippling issue. It has been a long time since 32.0.0.89 when videos worked. This issue was posted 8 months ago. The large list of related posts is a clear indicator that we are all struggling with this.

**Comment by ajwfrost** [comment]
Hi - thanks for the comment. The issues with H.264 and AAC are caused by licensing restrictions i.e. we’ve had to take out the software decoders that were in the runtimes. We’ve been looking into alternative approaches and want to get support for this back into AIR, using platform-specific implementations.

**Comment by gustavosanabio27** [comment]
H264VideoStreamSettings also does not work for publishing a camera to an RTMFP videoconference. I believe it is for the same reasons. So, I'm stuck with the AIR SDK 31 for desktop. :(

**Comment by Geddo** [comment]
@ajwfrost Thanks for the reply. I know that FLV still work but since there is no FLV encoder anymore (thanks to Adobe) there is currently no possibility to create a new App for iOS containing videos. As for me, every project of us I can think of has at least one Video in it so I cannot rebuild any of my apps now. So could you PLEASE give us a timeline or something when there might be video support be available again?

**Comment by Geddo** [comment]
Any news on this maybe? I can also provide  some beer if this helps :-)

**Comment by itlancer** [comment]
@ajwfrost 
Partially it works with latest AIR SDK 31.1.1.345 but it cannot be used in production because of:
Memory leaks (causing crashes):
https://github.com/Gamua/Adobe-Runtime-Support/issues/572
https://github.com/Gamua/Adobe-Runtime-Support/issues/587

Excessive memory consumption:
https://github.com/Gamua/Adobe-Runtime-Support/issues/646

Crashes with some videos:
https://github.com/Gamua/Adobe-Runtime-Support/issues/139

Also still have some related issues:
https://github.com/Gamua/Adobe-Runtime-Support/issues/224
https://github.com/Gamua/Adobe-Runtime-Support/issues/155

With macOS it still doesn't work for MP4 H.264 video with audio.

**Comment by itlancer** [comment]
@ajwfrost 
Test one more time. With macOS and AIR SDK 33.1.1.345 videos starts to play but it unusable because of https://github.com/Gamua/Adobe-Runtime-Support/issues/572
The same problems with Windows and macOS with latest AIR SDK 33.1.1.406.

**Comment by itlancer** [comment]
@ajwfrost 
With latest AIR SDK 33.1.476 for Windows video playback again works fine.
But with macOS there is still critical memory leak https://github.com/airsdk/Adobe-Runtime-Support/issues/572

**Comment by maxpropp** [comment]
Is there any update with this issue? I am using 33.1.1.444 and it is still having issues with some videos not playing and crashing on windows, but not on the IPad. 

33.1.476 does not seem to be a solution since it is having issues to compile for some reason I have not figured out.

**Comment by ajwfrost** [comment]
@maxpropp are you able to provide a sample video file if you're still seeing problems with not playing, or crashing?
Feel free to elaborate on the compilation issues, I don't believe we've heard of problems with that...!
thanks

**Comment by maxpropp** [comment]
> @maxpropp are you able to provide a sample video file if you're still seeing problems with not playing, or crashing?
> Feel free to elaborate on the compilation issues, I don't believe we've heard of problems with that...!
> thanks

@ajwfrost  this question was posted before the bug post. https://github.com/airsdk/Adobe-Runtime-Support/issues/857

**Comment by ajwfrost** [comment]
@maxpropp thanks - sorry, I hadn't tied those two together! Good to know it's just that one then :-)

**Comment by itlancer** [comment]
Finally most critical video issues fixed with latest AIR 33.1.1.575.
Thanks!
Hope other related video/audio issues will be fixed soon too:
https://github.com/airsdk/Adobe-Runtime-Support/issues/139
https://github.com/airsdk/Adobe-Runtime-Support/issues/155
https://github.com/airsdk/Adobe-Runtime-Support/issues/211
https://github.com/airsdk/Adobe-Runtime-Support/issues/224
https://github.com/airsdk/Adobe-Runtime-Support/issues/365
https://github.com/airsdk/Adobe-Runtime-Support/issues/89
https://github.com/airsdk/Adobe-Runtime-Support/issues/196
https://github.com/airsdk/Adobe-Runtime-Support/issues/646
https://github.com/airsdk/Adobe-Runtime-Support/issues/88
https://github.com/airsdk/Adobe-Runtime-Support/issues/1001
https://github.com/airsdk/Adobe-Runtime-Support/issues/151
https://github.com/airsdk/Adobe-Runtime-Support/issues/100
https://github.com/airsdk/Adobe-Runtime-Support/issues/93
https://github.com/airsdk/Adobe-Runtime-Support/issues/87
